### DISCUSSION TITLE
Filters - add radio button option for multi-select, so you don't have to click apply

### DISCUSSION THREAD

**Discussion by bryllyg** [microsoft-search/pnp-modern-search#1627] [Ideas]
**Is your feature request related to a problem? Please describe.**
From our usability studies, users are always missing or forgetting to click Apply after selecting items from a multi-value refinement item Refiner Template. It also confuses them that the apply link shows up at the top and sometimes the bottom.

**Describe the solution you'd like**
I would love to see radio buttons used here, and have the filter refresh dynamically on select, with no need to click apply.

**Describe alternatives you've considered**
No real option other than not to allow multi-select.

**Additional context**

**Comment by wobba** [comment]
Issue with radio and dynamic update is that you cannot accomplish an OR between the filters. And radio buttons are not meant as multi-select. Ever tried mashing two buttons at the same time on an old radio? 😉

A static filter with dynamic update on check box click is a lot easier.

**Comment by bryllyg** [comment]
Was copying that request from feedback I received and totally didn't notice the radio button miss-match. Check boxes, multi-select check boxes are the way to go, with dynamic updates. That would be ideal.

**Comment by tomobak** [comment]
I was about posting the same question...  Can we have dynamic filter for "Multi-value refinement item" without Apply To button?

**Comment by FranckyC** [comment]
Hi all, actually it would be worse in my opinion mainly for two reasons:

- SharePoint refiner values are dynamic by nature and calculated from the existing result set. It means for example, if a value is selected by an user at a specific moment and the query sent directly, refiner values could change between this moment you clicked on the value and the moment you're going to click on an other value in in initial list. That would be very annoying trust me.

- The other reason is about performances. When you have a large set of data, SharePoint search engine calls could be quite long to complete. Sending a request every time a value is selected would drastically decrease performances and result to UI inconsistencies described above. This would also introduce an unecessary complexity into the code to handle requests and UI correctly.

Sorry but we won't introduce this feature for now for these reasons.

**Comment by wobba** [comment]
I think it should be kept open of someone wants to take a stab at it, which is a bit painful in keeping state etc.

You can use the initial values coming back as static filters until you re-query. I've created similar filters in the past, but they only work if you know the finite set of values, which shouldn't be more than maybe 15?

In general I do agree with your points @FranckyC 🙂

**Comment by wobba** [comment]
Whoever picks this one up has to come up with a good plan on how to achieve it, or support it via a custom component.
